src/Microsoft.Health.Fhir.Api/Features/Filters/ValidateExportHeadersFilterAttribute.cs

```diff
@@ -0,0 +1,43 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using Hl7.Fhir.Rest;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.Health.Fhir.Api.Features.Filters
+{
+    /// <summary>
+    /// A filter that validates the Accept and Prefer headers in the request and short-circuits the pipeline if they are invalid.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method)]
+    internal class ValidateExportHeadersFilterAttribute : ActionFilterAttribute
+    {
+        private const string PreferHeaderName = "Prefer";
+        private const string PreferHeaderExpectedValue = "respond-async";
+
+        public override void OnActionExecuting(ActionExecutingContext context)
+        {
+            EnsureArg.IsNotNull(context, nameof(context));
+
+            if (!context.HttpContext.Request.Headers.TryGetValue(HeaderNames.Accept, out var acceptHeaderValue) ||
+                acceptHeaderValue.Count != 1 ||
+                !string.Equals(acceptHeaderValue[0], ContentType.JSON_CONTENT_HEADER, StringComparison.OrdinalIgnoreCase))
+            {
+                throw new RequestNotValidException(string.Format(Resources.UnsupportedHeader, HeaderNames.Accept));
```

Comment:
The message might be a bit misleading? It's not that the Accept header is not supported but more like invalid values specified?